From 517c97cb25f4af048159877bfd402363151aa8d8 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:07 -0500
Subject: [drm] nouveau/ltc: fix tag base address getting truncated if above 4GiB

Message-id: <1418852380-13061-57-git-send-email-rclark@redhat.com>
Patchwork-id: 102185
O-Subject: [RHEL7 drm 56/89] drm/nouveau/ltc: fix tag base address getting truncated if above 4GiB
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: 3.17

commit 2d108344d3cb8c8de9cf5afe5bced3b8a5796c98

Author:     Ben Skeggs <bskeggs@redhat.com>
AuthorDate: Tue Aug 12 09:27:54 2014 +1000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:45:31 2014 -0500

    drm/nouveau/ltc: fix tag base address getting truncated if above 4GiB

    commit 147ed897e807e1a6e2c6fd9dc91830f1d2831d43 upstream

    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/nouveau/core/subdev/ltcg/gf100.c b/drivers/gpu/drm/nouveau/core/subdev/ltcg/gf100.c
index f2f3338..effe888 100644
--- a/drivers/gpu/drm/nouveau/core/subdev/ltcg/gf100.c
+++ b/drivers/gpu/drm/nouveau/core/subdev/ltcg/gf100.c
@@ -135,7 +135,7 @@ gf100_ltcg_init_tag_ram(struct nouveau_fb *pfb, struct gf100_ltcg_priv *priv)
  if (ret) {
   priv->num_tags = 0;
  } else {
-  u64 tag_base = (priv->tag_ram->offset << 12) + tag_margin;
+  u64 tag_base = ((u64)priv->tag_ram->offset << 12) + tag_margin;
 
   tag_base += tag_align - 1;
   ret = do_div(tag_base, tag_align);
-- 
1.7.1